### PR TITLE
JBPM-9659: Upgrade deployments to WildFly 23

### PR DESCRIPTION
**Pull request by jesuino**
**JIRA**: 

[JBPM-9659](https://issues.redhat.com/browse/JBPM-9659)

**Related PRs*: 

https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1693
https://github.com/kiegroup/kie-soup/pull/216
https://github.com/kiegroup/droolsjbpm-knowledge/pull/543
https://github.com/kiegroup/drools/pull/3757
https://github.com/kiegroup/optaplanner/pull/1463
https://github.com/kiegroup/lienzo-tests/pull/124
https://github.com/kiegroup/appformer/pull/1163
https://github.com/kiegroup/kie-uberfire-extensions/pull/122
https://github.com/kiegroup/jbpm/pull/1974
https://github.com/kiegroup/kie-jpmml-integration/pull/62
https://github.com/kiegroup/droolsjbpm-integration/pull/2539
https://github.com/kiegroup/kie-wb-playground/pull/115
https://github.com/kiegroup/kie-wb-common/pull/3660
https://github.com/kiegroup/drools-wb/pull/1506
https://github.com/kiegroup/jbpm-designer/pull/913
https://github.com/kiegroup/jbpm-work-items/pull/212
https://github.com/kiegroup/jbpm-wb/pull/1506
https://github.com/kiegroup/optaplanner-wb/pull/399
https://github.com/kiegroup/kie-wb-distributions/pull/1115
https://github.com/kiegroup/openshift-drools-hacep/pull/116
https://github.com/kiegroup/process-migration-service/pull/5
https://github.com/kiegroup/optaweb-employee-rostering/pull/662
https://github.com/kiegroup/optaweb-vehicle-routing/pull/535

### CODE REVIEW

File: .gitignore
Comment (ge0ffrey) [inline .gitignore:12]: -1 to merge this.

This has side effects: it ignores for example `/foo/src/main/resources/bar/example.iml` too.
That might be acceptable, but in any case this belongs in a a separate PR, in a separate discussion and if we do end up doing it, we need to do it **consistently** for all .gitignore files.

**Comment by ge0ffrey** [comment]
If the .gitignore changes is rolled back, this looks good.

@jesuino Would you mind using the draft button instead of "do not merge" to signal it's not ready to merge yet?

File: .gitignore
Comment (tiagobento) [reply .gitignore:12]: Hi @ge0ffrey! Thx for looking at this PR. I added that because when I opened `optaplanner` on IntelliJ, I ended up with many "added" .iml files, which is not something we want.. I think. Other repositories have one .gitignore per maven module, always excluding `*.iml`. Idk what's the preferred approach here, but I'm assuming we don't want to commit `*.iml` files, right?

Anyway, I can revert those changes for this PR, of course.

**Comment by tiagobento** [comment]
@ge0ffrey This PR is part of a big set of PRs. We're not sure if making it "Draft" will get in the way of our CDB/FDB builds on `droolsjbpm-build-bootstrap`. If that's not too bad, we would like to keep it as is.

File: optaplanner-persistence/optaplanner-persistence-jaxb/pom.xml
Comment (rsynek) [inline optaplanner-persistence/optaplanner-persistence-jaxb/pom.xml:69]: This PR mostly changes `javax` artifacts to their `jakarta` alternatives. Is there a reason why this particular dependency moves in the opposite direction?
The `jakarta.json` seems to be more recent according to https://mvnrepository.com/artifact/org.glassfish/jakarta.json and https://mvnrepository.com/artifact/org.glassfish/javax.json.

File: .gitignore
Comment (jesuino) [reply .gitignore:12]: Hello @ge0ffrey would you please let us know your thoughts about TIago's comment?

Thanks!

Comment (tiagobento) [reply .gitignore:12]: Ok, I'll revert the changes.

**Review by mareknovotny — APPROVED** [review]

**Comment by mareknovotny** [comment]
jenkins retest this please

**Review by rsynek — APPROVED** [review]
Good to merge when at least the "Linux - Pull Request" build is green.

**Comment by tiagobento** [comment]
@rsynek Thanks for the review! FYI, we're also triggering CDB and FDB builds on kiegroup/droolsjbpm-build-bootstrap#1693. It builds and tests all the repositories.

**Comment by rsynek** [comment]
@tiagobento ok, I was simply referring to the fact that GHA does not see the new dependencyManagement entries outside this repository. The Jenkins build should, though.

**Review by tomasdavidorg — APPROVED** [review]

**Review by triceo — APPROVED** [review]
The PR is good to go on our end. Geoffrey's review is not coming, he is on PTO.

**Pull request merged**
